packages/main/src/ColorPicker.js

```diff
@@ -0,0 +1,531 @@
+import Input from "./Input.js";
+import Slider from "./Slider.js";
+import Label from "./Label.js";
+
+// Styles
+import ColorPickerCss from "./generated/themes/ColorPicker.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-color-picker",
+	properties: /** @lends sap.ui.webcomponents.main.ColorPicker.prototype */ {
+
+		/**
+		 * Defines the currently selected color of the <code>ui5-color-picker</code>.
+		 * *Note*:  If you need to have initially set color, or change the color programmatically, use this property
+		 * *Important*: Use RGBA format when modifying this property.
+		 * @type {String}
+		 * @public
+		 */
+		color: {
+			type: String,
+			defaultValue: "rgba(255, 0, 0, 1)",
+		},
+
+		/**
+		 * Defines the HEX code of the currently selected color
+		 * *Note*: If Alpha(transperancy) is set it is not included in this property. Use <code>color</code> property.
+		 * @type {String}
+		 * @public
+		 */
+		hex: {
+			type: String,
+			defaultValue: "000000",
+		},
+
+		/**
+		 * Defines the current main color which is selected via the hue slider and is shown in the main color square.
+		 * @type {String}
+		 * @private
+		 */
+		_mainColor: {
+			type: Object,
+		},
+
+		// Defines the currenty selected color from the main color section.
+		_color: {
+			type: Object,
+		},
+
+		/**
+		 * @private
+		 */
+		_selectedCoordinates: {
+			type: Object,
+		},
+
+		/**
+		 * @private
+		 */
+		_alpha: {
+			type: Float,
+			defaultValue: 1,
+		},
+
+		/**
+		 * @private
+		 */
+		_hue: {
+			type: Integer,
+			defaultValue: 0,
+		},
+
+		/**
+		 * @private
+		 */
+		_wrongHEX: {
+			type: Boolean,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.ColorPicker.prototype */ {
+		//
+	},
+	events: /** @lends sap.ui.webcomponents.main.ColorPicker.prototype */ {
+		change: {},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * For the <code>ui5-color-picker</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/ColorPicker.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @since 1.0.0-rc12
+ * @alias sap.ui.webcomponents.main.ColorPicker
+ * @extends UI5Element
+ * @tagname ui5-color-picker
+ * @public
+ */
+class ColorPicker extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return ColorPickerCss;
+	}
+
+	static get template() {
+		return ColorPickerTemplate;
+	}
+
+	static get dependencies() {
+		return [
+			Input,
+			Slider,
+			Label,
+		];
+	}
+
+	constructor() {
+		super();
+
+		// Top Left corner
+		this._selectedCoordinates = {
+			x: -6.5,
+			y: -6.5,
+		};
+
+		// Default color is red
+		this._mainColor = {
+			r: 255,
+			g: 0,
+			b: 0,
+		};
+
+		this._color = {
+			r: 0,
+			g: 0,
+			b: 0,
+		};
+
+		this.mouseDown = false;
+	}
+
+	onBeforeRendering() {
```

Comment:
This will be executed on every rerender, while it's actually only necessary upon user input (when the user changes colors). Furthermore, you can't fire the "change" event here once you implement it.

The best flow is: upon user interaction, in the event handler of the respective DOM event, change the public state (color here) and fire the change event. During onBeforeRendering only calculate derived state.